src/main/java/com/minecolonies/colony/permissions/Permissions.java

```diff
@@ -255,121 +236,102 @@ public void savePermissions(NBTTagCompound compound)
     /**
      * Returns a set of UUID's that have permission to send (and receive) messages
      *
-     * @return      Set of UUID's allowed to send and receive messages
+     * @return Set of UUID's allowed to send and receive messages
      */
     public Set<UUID> getMessagePlayers()
     {
         return players.values().stream().filter(player ->
-                hasPermission(player.rank, Action.SEND_MESSAGES)).map(player -> player.id).collect(Collectors.toSet());
-
-    }
-
-    public Set<Player> getPlayersByRank(Rank rank)
-    {
-        return this.players.values().stream()
-                           .filter(player -> player.rank.equals(rank))
-                           .collect(Collectors.toSet());
-    }
-
-    public Set<Player> getPlayersByRank(Set<Rank> ranks)
-    {
-        return this.players.values().stream()
-                           .filter(player -> ranks.contains(player.rank))
-                           .collect(Collectors.toSet());
+                                                  hasPermission(player.rank, Action.SEND_MESSAGES)).map(player -> player.id).collect(Collectors.toSet());
     }
 
     /**
-     * Returns the map of permissions and ranks
+     * Checks if a rank can perform an action
      *
-     * @return          map of permissions
+     * @param rank   Rank you want to check
+     * @param action Action you want to perform
+     * @return true if rank has permission for action, otherwise false
      */
-    public Map<Rank, Integer> getPermissions()
+    public boolean hasPermission(Rank rank, @NotNull Action action)
     {
-        return permissions;
+        return Utils.testFlag(permissions.get(rank), action.flag);
     }
 
-    /**
-     * Returns the rank of a player
-     *
-     * @param player    player to check rank
-     * @return          Rank of te player
-     */
-    public Rank getRank(EntityPlayer player)
+    public Set<Player> getPlayersByRank(Rank rank)
     {
-        return getRank(player.getGameProfile().getId());
-    }
-    /**
+        return this.players.values().stream()
+                 .filter(player -> player.rank.equals(rank))
+                 .collect(Collectors.toSet());
+    }    /**
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ATrailingCommentCheck)